Description: Support centering the gmrun window
 If the user hasn't set any position prefs (Top or Left directives), then
 center the window
Bug-Debian: https://bugs.debian.org/471319
Author: Vincent Legout <vincent@legout.info>

--- a/src/main.cc
+++ b/src/main.cc
@@ -629,8 +629,8 @@
 
   gtk_box_pack_start(GTK_BOX(hbox), compline, TRUE, TRUE, 0);
   
-  int prefs_top = 80;
-  int prefs_left = 100;
+  int prefs_top = -1;
+  int prefs_left = -1;
   configuration.get_int("Top", prefs_top);
   configuration.get_int("Left", prefs_left);  
 
@@ -659,10 +659,14 @@
     geo_parsed = gtk_window_parse_geometry (GTK_WINDOW (win),
 					    geoptr);
   }
-  else
+  else if (prefs_top != -1 && prefs_left != -1)
   {
     gtk_widget_set_uposition(win, prefs_left, prefs_top);
   }
+  else
+  {
+    gtk_window_set_position (GTK_WINDOW (win), GTK_WIN_POS_CENTER);
+  }
 
   gtk_widget_show(win);
 
